### DISCUSSION TITLE
I add a new ST7701s setting and init codes for a beautiful 480x800 4inch None-Air-Gap IPS screen. Need help to optimize it.

### DISCUSSION THREAD

**Discussion by hugo8873** [moononournation/Arduino_GFX#366] [General]
Hello all,
Im very new to Github and recently I test a none air gap IPS touch screen hardware, want to share what I got here with everyone.
(I need help on take a second look at my codes and wiring, since I grab codes from here and there, and some of them I don't really understand, would you take a moment to look at my init code for this screen, if there anything not so optimized?  Such as the pins, data bus, etc. Appreciated.)
(And I also want to retrieve some pins for more peripherals, any idea remove some unnecessary pins?)

Screen bought from https://item.taobao.com/item.htm?spm=a1z09.2.0.0.319a2e8dE909QS&id=614606266642&_u=q1su9gba322
Having a ST7701S with 45-pin (3-SPI  + RBG565) FPC connector
Touch IC: GT911
![IMG_4214](https://github.com/moononournation/Arduino_GFX/assets/22760386/f50aa8fb-56a6-4edd-9582-c912766ecb37)

https://github.com/moononournation/Arduino_GFX/assets/22760386/9ccbbd28-501e-46b6-ae51-243f5c4673ec

I add a new st7701_type9_init_operations[] with help from the manufacturer
[static const uint8_t st7701_type9_init_operations[].txt](https://github.com/moononournation/Arduino_GFX/files/13172985/static.const.uint8_t.st7701_type9_init_operations.txt)
Add this section of code inside ~/src/display/Arduino_RGB_Display.h

Also attaching a screen data sheet.
[KD040WVFPD025-C008B SPEC V1.0 .pdf](https://github.com/moononournation/Arduino_GFX/files/13172992/KD040WVFPD025-C008B.SPEC.V1.0.pdf)

And a init code for Arduino IDE, running on ESP32-S3-Devkit-C1 N16R8
It can run with LVGL(SquarelineStudio)

`#define GFX_BL 46

Arduino_DataBus *bus = new Arduino_SWSPI(GFX_NOT_DEFINED /* DC */, 10 /* CS */, 12 /* SCK */, 11 /* MOSI */, GFX_NOT_DEFINED /* MISO */);

Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel(
  45 /* DE */, 4 /* VSYNC */, 5 /* HSYNC */, 21 /* PCLK */,
  39 /* R0 */, 40 /* R1 */, 41 /* R2 */, 42 /* R3 */, 2 /* R4 */,
  0 /* G0 */, 9 /* G1 */, 14 /* G2 */, 47 /* G3 */, 48 /* G4 */, 3 /* G5 */,
  6 /* B0 */, 1 /* B1 */, 13 /* B2 */, 16 /* B3 */, 8 /* B4 */,
  1 /* hsync_polarity */, 10 /* hsync_front_porch */, 8 /* hsync_pulse_width */, 50 /* hsync_back_porch */,
  1 /* vsync_polarity */, 10 /* vsync_front_porch */, 8 /* vsync_pulse_width */, 20 /* vsync_back_porch */,
  1 /* pclk_active_neg */, 16000000 /* prefer_speed */, false /*useBigEndian*/,
  0 /*de_idle_high*/, 1 /*pclk_idle_high*/);

Arduino_RGB_Display *gfx = new Arduino_RGB_Display(
  480 /* width */, 800 /* height */, rgbpanel, 0 /* rotation */, true /* auto_flush */,
  bus, 38 /* RST */, st7701_type9_init_operations, sizeof(st7701_type9_init_operations));`

In addition, SPI pins:
 #define TOUCH_GT911_SCL 43
 #define TOUCH_GT911_SDA 44
 #define TOUCH_GT911_INT 7
 #define TOUCH_GT911_RST 15

And Hi @moononournation ,
Do you think it's worthwhile to add this new ST7701s type9 into your repository?

**Comment by moononournation** [comment]
Sure, please raise a pull request with a PDQgraphicstest Youtube link
